crates/runtime/src/dataconnector/listing/connector.rs

```diff
@@ -765,16 +769,47 @@ fn to_listing_table_url(
     path: &Path,
     dataset: &Dataset,
     dataconnector: &str,
-) -> DataConnectorResult<ListingTableUrl> {
+) -> DataConnectorResult<SensitiveListingTableUrl> {
     let mut new_url = original_url.clone();
     new_url.set_path(&format!("/{path}"));
 
-    ListingTableUrl::parse(new_url).boxed().context(
+    let sensitive_url = ListingTableUrl::parse(&new_url).boxed().context(
         crate::dataconnector::UnableToGetSchemaInternalSnafu {
             dataconnector: dataconnector.to_string(),
             connector_component: ConnectorComponent::from(dataset),
         },
-    )
+    )?;
+
+    Ok(SensitiveListingTableUrl::new(sensitive_url, new_url))
+}
+
+fn sanitize_url(mut url: Url) -> Url {
+    url.set_fragment(None);
```

Comment:
Fragment is the only place we'd expect secrets?

Reply:
Yes, we add them to the fragment for the object store registry to extract